BasisSetModule.jl

```diff
@@ -0,0 +1,635 @@
+"NIHONIUM" =>                "Nh", 
+"FLEROVIUM" =>               "Fl", 
+"MOSCOVIUM" =>               "Mc", 
+"LIVERMORIUM" =>             "Lv", 
+"TENNESSINE" =>              "Ts", 
+"OGANESSON" =>               "Og", 
+)
+
+# Supplementary Functions for the PEG BasisSet Parsers
+
+
+function dict(children) 
+  dicty = Dict() 
+for i = 1:length(children) 
+  dicty = merge(dicty,children[i]) 
+end 
+
+return dicty
+end
+
+function DictElements(children)
+Elem = children[1][1]
+contracteds = Vector()
+for i=1:length(children)
+ cont = BasisSetModule.ContractedGaussianBasisFunctionDefinition(children[i][2],children[i][3][1])
+ push!(contracteds,cont)
+end
+
+dictyElem = Dict(zip([Elem],[contracteds]))
+
+return dictyElem
+end
+
+function bases(children)
+bases = Vector()
+for i=1:length(children)
+if typeof(children[i]) == QuantumLab.BasisSetModule.ContractedGaussianBasisFunctionDefinition
+  push!(bases,children[i])
+elseif typeof(children[i]) == Array{QuantumLab.BasisSetModule.ContractedGaussianBasisFunctionDefinition,1}
+  for j=1:length(children[i])
+  push!(bases,children[i][j]) 
+  end
+end
+end
+
+return bases
+end
+
+function SPOrbitals(children)
+OrbS = Vector()
+OrbP = Vector()
+S = Vector()
+P = Vector()
+for i=1:length(children)
+  S1 = children[i][1]
+  push!(S,S1)
+end
+for i=1:length(children)
+  S2 = children[i][2]
+  push!(P,S2)
+end
+
+OrbS = BasisSetModule.ContractedGaussianBasisFunctionDefinition(LQuantumNumber("S"),S)
+OrbP = BasisSetModule.ContractedGaussianBasisFunctionDefinition(LQuantumNumber("P"),P)
+#push!(Primitives,OrbS)
+#push!(Primitives,OrbP)
+
+
+Primitives = vcat(OrbS,OrbP)
+return Primitives
+end
+
+function dictDalton(children) 
+  dicty = Dict() 
+for i = 1:length(children)
+  dicty = merge(dicty,children[i])
+end 
+
+return dicty
+end
+
+
+function basesDalton(children)
+bases = Vector()
+for i=1:length(children)
+  for j=1:length(children[i])
+  push!(bases,children[i][j]) 
+  end
+end
+
+return bases
+end
+
+
+function CreateCorrectPrimitives(children)
+Primitives = Vector()
+A = zeros(length(children[2]),length(children[2][1]))
+for i = 1:length(children[2]) 
+  for j = 1:length(children[2][i])
+  A[i,j] = children[2][i][j]
+  end
+end 
+B = 0
+
+a = size(A)
+C = [Vector{Any}() for _ in 1:a[2]-1]
+
+for j = 1:a[2]-1
+  for i = 1:a[1]
+    if A[i,j] == 0 
+    continue
+    else 
+    B = BasisSetModule.PrimitiveGaussianBasisFunctionDefinition(A[i,end],A[i,j])
+    push!(C[j],B)
+    end
+  end
+end 
+
+
+
+D = Vector(length(C))
+for i=1:length(C)
+  D[i]=BasisSetModule.ContractedGaussianBasisFunctionDefinition(children[1],C[i])
+end
+return D
+end
+
+function LinearFactors(children)
+
+LinearFactor=0
+for i=1:length(children)
+  if children[i][1] == 0
+  continue
+  else 
+  LinearFactor=children[i][1]
+  end
+end
+return LinearFactor
+end
+############################################################################################################################################
+# Main Functions
```

Comment:
I'd like to go over these functions in detail. This will probably be much easier once you have made the other adjustments, so I'll wait until then. Same holds for `runtests.jl` below. Thank you for updating your pull request (simply push your changes to sasadada:master).